src/api/pinned.ts

```diff
@@ -0,0 +1,78 @@
+// pinned.ts - pinned tickets methods
+
+import { AxiosInstance } from "axios";
+import { TicketRecord } from "@/api/tickets";
+
+export interface PinnedTicketRecord {
+  ticket: TicketRecord;
+  pinned: string; // date time
+  object_uuid: string;
+  created: string; // date
+  id: string; // pk
+}
+
+export async function getPinnedTicketsForMember(
+  axios: AxiosInstance,
+  teamId: number,
+  memberPk: string
+) {
+  const response = await axios.get(
+    `/api/teams/${teamId}/members/${memberPk}/pinned_tickets/`
+  );
+
+  console.log(response);
+
+  return response.data as PinnedTicketRecord[];
+}
+
+export async function deletePinnedTicket(
+  axios: AxiosInstance,
+  teamId: number,
+  memberPk: string,
+  pinnedTicketPk: string
+) {
+  await axios.delete(
+    `/api/teams/${teamId}/members/${memberPk}/pinned_tickets/${pinnedTicketPk}/`
+  );
+
+  return;
+}
+
+export async function postPinnedTicket(
+  axios: AxiosInstance,
+  teamId: number,
+  memberPk: string,
+  ticketId: number
+): Promise<PinnedTicketRecord> {
+  const response = await axios.post(
+    `/api/teams/${teamId}/members/${memberPk}/pinned_tickets/`,
+    {
+      ticket: ticketId
+    }
+  );
+
+  return response.data as PinnedTicketRecord;
+}
+
+export async function getPinnedTicketForTicket(
+  axios: AxiosInstance,
+  teamId: number,
+  memberPk: string,
+  ticketId: number
+): Promise<PinnedTicketRecord> {
+  const pinnedTickets = await getPinnedTicketsForMember(
```

Comment:
This function seems unnecessary. The api can be set up to filter based on query params and I think it would cut down on api calls. This functionality might not yet be available, so it's understandable that you wrote something like this. Keep it for now, but in the future we should move the filtering functionality to the backend.